### PR TITLE
[MM-40681] : The "Leave Channel" button doesn't work from the 3-dot menu when clicking on it second time

### PR DESCRIPTION
**Pull request by codedsun**
#### Summary

Fixed the callback from the function of `getCloseHandler`

#### Ticket Link

Jira ticket: https://mattermost.atlassian.net/browse/MM-40681

Fix https://github.com/mattermost/mattermost-server/issues/19209

 Fixes https://github.com/mattermost/mattermost-server/issues/40681

#### Release Note

```
NONE
```

### CODE REVIEW

**Comment by codedsun** [comment]
@AGMETEOR Review please

**Comment by M-ZubairAhmed** [comment]
removing @devinbinnie since he is on 🎄 , adding myself

File: components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx [hunk 1]
```diff
@@ -22,19 +22,20 @@ interface Props extends PropsFromRedux {
...
     }
 
-    handleLeavePrivateChannel = () => {
+    handleLeavePrivateChannel = (callback: () => void) => {
         this.props.actions.openModal({modalId: ModalIdentifiers.LEAVE_PRIVATE_CHANNEL_MODAL, dialogType: LeavePrivateChannelModal, dialogProps: {channel: this.props.channel}});
         trackEvent('ui', 'ui_private_channel_x_button_clicked');
+        callback();
     }
 
     getCloseHandler = () => {
```
Comment (AGMETEOR) [inline components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: Could we add a unit test for this component's instance to ensure getCloseHandler works well?

Comment (AGMETEOR) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: Hello @codedsun . Do you need any help on this?

Comment (codedsun) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: @AGMETEOR - Yes I am a little confused to where to write the unit test and how?

Comment (AGMETEOR) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: This would be the file with the file https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx where to put the test. It's just next to the file where you made the changes.

See example here on how to test a method inside a class component https://stackoverflow.com/a/41196670/7723621

Comment (AGMETEOR) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: > This would be the file with the file https://github.com/mattermost/mattermost-webapp/blob/master/components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx where to put the test. It's just next to the file where you made the changes.
> 
> See example here on how to test a method inside a class component https://stackoverflow.com/a/41196670/7723621

Let me know if you need more help.

Comment (codedsun) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: @AGMETEOR Let me try today and if I can't do it, will ping you and connect for a call. Thanks

Comment (codedsun) [reply components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: @AGMETEOR Done 💯

**Comment by codedsun** [comment]
@M-ZubairAhmed @AGMETEOR Review please!

File: components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx [hunk 2]
```diff
@@ -91,4 +91,25 @@ describe('components/sidebar/sidebar_channel/sidebar_base_channel', () => {
...
+        const callback = jest.fn();
+        const props = {
+            ...baseProps,
+            channel: {
+                ...baseProps.channel,
+                type: 'P' as ChannelType,
+            },
+        }
+        const wrapper = shallow<SidebarBaseChannel>(<SidebarBaseChannel {...props} />);
+        wrapper.instance().handleLeavePublicChannel(callback);
```
Comment (M-ZubairAhmed) [inline components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.test.tsx:112]: Shouldn't this be handleLeavePrivate... ?

File: components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx [hunk 3]
```diff
@@ -22,19 +22,20 @@ interface Props extends PropsFromRedux {
...
         const {channel} = this.props;
-
```
Comment (M-ZubairAhmed) [inline components/sidebar/sidebar_channel/sidebar_base_channel/sidebar_base_channel.tsx:37]: Nit: unrelated change, could you please revert

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by AGMETEOR — APPROVED** [review]
@codedsun LGTM! Please address @M-ZubairAhmed 's comments and all is good. Thanks a lot for the contribution 🎉

**Comment by codedsun** [comment]
@M-ZubairAhmed Done the changes.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @codedsun 
Tested, looks good to merge.
- Verified Leave channel, a sidebar 3-dot menu option, is working as expected.
- [E2E report ](https://automation-dashboard.vercel.app/cycles/fb4ce4ec-5456-4113-8e90-3fcf9b1dc802) looks good, no PR related failures.
